code-challenges.test.js

```diff
@@ -15,48 +20,117 @@
+        //     return string.replace(/[iI]/g, 1)
+        // } else if (x = 'o') {
+        //     return string.replace(/[oO]/g, 0)
+        // }
+            
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
-
+describe('only_letter', () => {
 var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
 var letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
 var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
 var letterE = "e"
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter.", () => {
+        expect(only_letter(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"]),
+        expect(only_letter(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+    // Expected output: ["Apple", "Banana", "Orange"]
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
-
-
 // b) Create the function that makes the test pass.
-
-
+//create a function taking in an array and string
+//iterate with .filter
+//see if the word contains specific letter
+//use .toUpperCase() to match correct letter
+const only_letter = (array, string) => {
+    return array.filter(value => value.toUpperCase().includes(string.toUpperCase())
+    )
+}
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
+describe("fullHouse", () => {
 
 var hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
 var hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
 var hand3 = [5, 5, 5, 5, 4]
+it("takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.", () => {
+    expect(fullHouse(hand1)).toEqual(true)
+    expect(fullHouse(hand2)).toEqual(false)
+    expect(fullHouse(hand3)).toEqual(false)
+  })
+})
+// Expected output: true
+// Expected output: false
 // Expected output: false
 
 
 
 // b) Create the function that makes the test pass.
+//create function that takes in an array
+//creating an empty object to hold the count of each value
+// of will take in an array and return with no properties.
+//define card to become the values in the empty object.
+//iterate checking for matches
+//I found this example on stack overflow with no explination. I personally used https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of to learn about for..of.
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values for information on Object.values() which i dont 100% understand yet.
```

Comment:
You're not alone in using this approach! A lot of people grab code in their careers they don't understand which is okay until you need to know what to do with it. It's important to reach out to a mentor or a friend who can walk you through something like this!

Feel free to reach out to me and bring this question to office hours